### PR TITLE
Create a shared ESLint config

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

With this PR we are establishing a new home for shared npm packages, a multi-package repository managed by [Lerna](https://lerna.js.org/). For now there is just one package here, the shared ESLint configuration (discussed below), but the intention is that we will import the design system library into this repo as well and decommission the `case-triage` repo. This will maintain continuity for development and automated deploy tooling on the `case-triage` repo in the meantime while we bring this up to feature parity with automated deploys and whatnot (which Lerna should be helpful for). 

The "repo establishment" portion of this PR also includes the usual new repo boilerplate: documentation, PR and issue templates, etc.

Anyway, on to the package itself: it's a synthesis of the configurations we were using in our various active frontend projects. Thankfully they are all based on one another and there were not major differences between them. I retained almost all of the overrides from the old configs, except a handful that seemed better left to the individual packages or that I just thought were unnecessary. There are draft PRs open where you can see what the effects would be in: 

- [recidiviz-data](https://github.com/Recidiviz/recidiviz-data/pull/6794) (admin panel and case triage frontends)
- [case-triage](https://github.com/Recidiviz/case-triage/pull/77) (aka the design system)
- [pulse-dashboard](https://github.com/Recidiviz/pulse-dashboard/pull/995)
- [spotlight](https://github.com/Recidiviz/public-dashboard/pull/386)

The dependency management was the trickiest part, because in some places we are competing with Create React App and there are lots of annoying peer dependencies in the ESLint plugin chain for ["reasons"](https://github.com/eslint/eslint/issues/3458). Basically if you don't install all of them in your dependent package then some rules may not be applied or you may get errors, it's unclear though because `react-scripts` throws off a ton of spurious peer dependency warnings that we tend to ignore. The good news is that you can pretty much copy the entire peer dependencies object into your package.json `devDependencies` and the yarn warnings will quiet down. (Older `react-scripts` installs will get warnings about the wrong version of ESLint but .... it seems fine?)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

part of https://github.com/Recidiviz/recidiviz-data/issues/6719

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by nbhargava — APPROVED** [review]
Into it

**Review by daschi — APPROVED** [review]
This is awesome! I like the name `web-libraries`, and the changes in `pulse` seem to be all good. `yalc` was also easy to use and worked locally.

**Review by jovergaag — APPROVED** [review]
Sweet!

I created an [issue](https://app.zenhub.com/workspaces/torch--5ec36552cdfe7bc58f04bdbe/issues/recidiviz/pulse-dashboard/996) for pulse-dashboard to migrate over to this config.

File: packages/eslint-config/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+```bash
+# Use the linked component library.
+# This will add some local config files that should be ignored.
+# It will also change your package.json and yarn.lock,
+# be sure not to commit these changes!
+$ yalc add @recidiviz/eslint-config
+# install any transitive dependencies;
+# only needed if you change them while developing
+$ yarn install
+```
```
Comment (Catacola) [inline packages/eslint-config/README.md:36]: Slick! I was afraid this was going to become a git-submodule rat's nest, but this seems pretty straightforward.

File: packages/eslint-config/index.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        "@typescript-eslint/no-shadow": "error",
+        "no-use-before-define": "off",
+        "@typescript-eslint/no-use-before-define": ["error"],
+        // TypeScript makes these redundant
+        "react/prop-types": "off",
+        "react/require-default-props": "off",
+      },
+    },
+    {
+      files: ["**.test.js"],
```
Comment (Catacola) [inline packages/eslint-config/index.js:113]: For the overly eager, should we add TS here as well?
```
files: ["**.test.js", "**.test.ts"],
```

**Comment by macfarlandian** [comment]
ok I think I have quorum for merge but feel free to open an issue or yell at me through the channel of your choosing if you find there are any rule choices you don't agree with

**Review by Catacola — APPROVED** [review]
Seems like a reasonable start.

File: packages/eslint-config/index.js (hunk 2 above)
Comment (macfarlandian) [reply packages/eslint-config/index.js:113]: hmm good callout, I think probably you are right. let me look into that quickly before merging

**Pull request merged**